homeassistant/components/smlight/sensor.py

```diff
@@ -0,0 +1,163 @@
+    ),
+    SmSensorEntityDescription(
+        key="fs_usage",
+        translation_key="fs_usage",
+        device_class=SensorDeviceClass.DATA_SIZE,
+        native_unit_of_measurement=UnitOfInformation.KILOBYTES,
+        entity_registry_enabled_default=False,
+        value_fn=lambda x: x.fs_used,
+    ),
+    SmSensorEntityDescription(
+        key="core_uptime",
+        translation_key="core_uptime",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_registry_enabled_default=False,
+        value_fn=lambda x: x.uptime,
+    ),
+    SmSensorEntityDescription(
+        key="socket_uptime",
+        translation_key="socket_uptime",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_registry_enabled_default=False,
+        value_fn=lambda x: x.socket_uptime if x.socket_uptime > 0 else None,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SmConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up SMLIGHT sensor based on a config entry."""
+    coordinator: SmDataUpdateCoordinator = entry.runtime_data
+
+    async_add_entities(
+        SmSensorEntity(coordinator, description) for description in SENSORS
+    )
+
+
+class SmSensorEntity(SmEntity, SensorEntity):
+    """Representation of a slzb sensor."""
+
+    def __init__(
+        self,
+        coordinator: SmDataUpdateCoordinator,
+        description: SmSensorEntityDescription,
+    ) -> None:
+        """Initiate slzb sensor."""
+        super().__init__(coordinator)
+
+        self.entity_description: SmSensorEntityDescription = description
+        self._attr_unique_id = f"{coordinator.unique_id}_{description.key}"
+        self._last_uptime: datetime | None = None
+
+    def get_uptime(self, uptime: float | None) -> datetime | None:
+        """Return device or zigbee socket uptime.
+
+        Converts uptime from seconds to a datetime value, allow up to 5
+        seconds deviation. Fire a reboot event if device uptime has reset
+        since last run.
+        """
+        if uptime is None:
+            # reset to unknown state
+            self._last_uptime = None
+            return None
+
+        delta = timedelta(seconds=uptime)
+
+        if "core_uptime" in self.entity_description.key and delta <= SCAN_INTERVAL:
+            self.fire_reboot_event()
+
+        new_uptime = utcnow() - delta
+
+        if (
+            not self._last_uptime
+            or abs(new_uptime - self._last_uptime) > UPTIME_DEVIATION
+        ):
+            self._last_uptime = new_uptime
+
+        return self._last_uptime
+
+    def fire_reboot_event(self):
+        """Fire reboot event."""
+        self.coordinator.hass.bus.async_fire(
+            SMLIGHT_SLZB_REBOOT_EVENT,
+            {
+                "device_id": self.coordinator.unique_id,
+                "host": self.coordinator.hostname,
+            },
+        )
+        LOGGER.debug("SLZB device reboot detected")
```

Comment:
I'd rather see an event entity displaying this

Reply:
sure, will do in uptime PR